src/main/java/uk/ac/cam/cl/dtg/isaac/api/QuizFacade.java

```diff
@@ -0,0 +1,192 @@
+            String message = "ContentManagerException whilst getting available quizzes";
+            log.error(message, e);
+            return new SegueErrorResponse(Response.Status.INTERNAL_SERVER_ERROR, message).toResponse();
+        }
+    }
+
+    /**
+     * Preview a quiz. Only available to teachers and above.
+     *
+     * @param request
+     *            - so we can deal with caching.
+     * @param httpServletRequest
+     *            - so that we can extract user information.
+     * @param quizId
+     *            as a string
+     * @return a Response containing a list of ContentSummaryDTO for the visible quizzes.
+     */
+    @GET
+    @Path("/{quizId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @GZIP
+    @ApiOperation(value = "Preview an individual quiz.")
+    public final Response previewQuiz(@Context final Request request, @Context final HttpServletRequest httpServletRequest,
+                                      @PathParam("quizId") final String quizId) {
+        try {
+            AbstractSegueUserDTO user = this.userManager.getCurrentUser(httpServletRequest);
+
+            if (!(isUserTeacherOrAbove(user))) {
+                throw new ForbiddenException("Students cannot preview quizzes.");
+            }
+
+            if (null == quizId || quizId.isEmpty()) {
+                return new SegueErrorResponse(Response.Status.BAD_REQUEST, "You must provide a valid quiz id.").toResponse();
+            }
+
+            // Calculate the ETag on current live version of the content
+            // NOTE: Assumes that the latest version of the content is being used.
+            EntityTag etag = new EntityTag(this.contentManager.getCurrentContentSHA().hashCode() + quizId.hashCode() + "");
+
+            Response cachedResponse = generateCachedResponse(request, etag);
+            if (cachedResponse != null) {
+                return cachedResponse;
+            }
+
+            IsaacQuizDTO quiz = this.quizManager.findQuiz(quizId);
+
+            Response cachableResult = Response.ok(quiz)
+                .cacheControl(getCacheControl(NUMBER_SECONDS_IN_ONE_HOUR, false)).tag(etag).build();
+            return cachableResult;
+        } catch (SegueDatabaseException e) {
+            String message = "SegueDatabaseException whilst previewing a quiz";
+            log.error(message, e);
+            return new SegueErrorResponse(Response.Status.INTERNAL_SERVER_ERROR, message).toResponse();
+        } catch (ContentManagerException e) {
+            String message = "ContentManagerException whilst previewing a quiz";
+            log.error(message, e);
+            return new SegueErrorResponse(Response.Status.INTERNAL_SERVER_ERROR, message).toResponse();
+        }
+    }
+
+    private static boolean isUserRegistered(AbstractSegueUserDTO user) {
+        return user instanceof RegisteredUserDTO;
+    }
+
+    private static boolean isUserTeacherOrAbove(AbstractSegueUserDTO user) {
```

Comment:
I added [`AbstractSegueFacade::isUserTeacherOrAbove`](https://github.com/isaacphysics/isaac-api/blob/v2.15.2/src/main/java/uk/ac/cam/cl/dtg/segue/api/AbstractSegueFacade.java#L326-L329) after seeing this and noticing the lack of checks in the Groups and Assignments facades. 
I know the AbstractFacade method is a little clunckier, but it can at least be used consistently that way. It just needs the usages of this method changing to `isUserTeacherOrAbove(userManager, user)`, noting that it needs to be a registered user though.

Reply:
If you swap to `getCurrentRegisteredUser` as mentioned above, this should work just fine!